http_proxy.go

```diff
@@ -45,28 +46,16 @@ func main() {
 		log.Error(err)
 	}
 
-	// Middleware (runs in reverse order as they are added)
-
-	// Middleware: Forward HTTP Messages
-	forwarder, err := forward.New(nil, forward.IdleTimeoutSetter(time.Duration(*idleClose)*time.Second))
-	if err != nil {
-		log.Error(err)
-	}
-
-	// Middleware: Handle HTTP CONNECT
-	httpConnect, err := httpconnect.New(forwarder, httpconnect.IdleTimeoutSetter(time.Duration(*idleClose)*time.Second))
-	if err != nil {
-		log.Error(err)
-	}
-
-	// Middleware: Common request filter
-	commonHandler, err := commonfilter.New(httpConnect, testingLocal)
-	if err != nil {
-		log.Error(err)
-	}
+	filterChain := filter.NewChain(
```

Comment:
Here you see how easy it is to set up a filter chain, with filters in the order that they actually get executed.